front/src/BottomBar/index.js

```diff
@@ -7,86 +7,149 @@ import faThLarge from "@fortawesome/fontawesome-free-solid/faThLarge";
-      </Button>
-      <Button
-        id="bottom-bar__stop-button"
-        onClick={() => onSend("stop")}
-        enabled
-      >
-        <FontAwesomeIcon icon={faStop} color="red" />
-      </Button>
+        <ConnectionStatus>
+          {connecting
+            ? connectionTypes.CONNECTING
+            : connected
+              ? connectionTypes.CONNECTED
+              : connectionTypes.DISCONNECTED}
+        </ConnectionStatus>
+      </ConnectionBar>
+      <ButtonsWrapper className={props.className}>
+        <ButtonColumn className={props.className}>
+          <Button
+            id="bottom-bar__start-button"
+            onClick={() => onSend("start")}
+            enabled
+          >
+            <FontAwesomeIcon icon={faPlay} />
+          </Button>
+          <Button
+            id="bottom-bar__change-page-button"
+            onClick={() =>
+              changeActivePage(
+                isSettingsPage ? pageTypes.REFBOX : pageTypes.SETTINGS
+              )
+            }
+            enabled
+          >
+            {isSettingsPage ? (
+              <FontAwesomeIcon icon={faThLarge} />
+            ) : (
+              <FontAwesomeIcon icon={faCog} />
+            )}
+          </Button>
+        </ButtonColumn>
+        <StopButton
+          id="bottom-bar__stop-button"
+          onClick={() => onSend("stop")}
+          enabled
+        >
+          <FontAwesomeIcon icon={faStop} color="red" />
+        </StopButton>
+      </ButtonsWrapper>
     </Bar>
   );
 };
 
 BottomBar.propTypes = {
   onSend: PropTypes.func.isRequired,
   changeActivePage: PropTypes.func.isRequired,
-  activePage: PropTypes.oneOf(Object.values(pageTypes))
+  activePage: PropTypes.oneOf(Object.values(pageTypes)),
+  isConnected: PropTypes.oneOf(Object.values(connectionTypes))
 };
 
-const Bar = styled.div`
-  position: fixed;
-  background-color: white;
-  bottom: 0;
-  width: 100%;
-  height: 10%;
-  border-style: solid;
-  border-width: 0.2rem 0rem 0rem 0rem;
+const ButtonsWrapper = styled.div`
   display: flex;
   justify-content: space-between;
   margin: 0rem;
+  flex: 1;
+`;
+
+const ButtonColumn = styled.div`
+  width: 50%;
+  height: 100%;
+  justify-content: space-between;
+  margin: 0rem;
+  flex: 1;
 `;
 
 const Button = styled.button`
   background: ${props => props.theme.bottomBarButton};
-  width: 32%;
-  font-size: 5rem;
+  width: 100%;
+  height: 50%;
+  font-size: 200%;
   flex: 1;
+  &:active {
+    background: ${props => props.theme.bottomBarButtonActive};
+  }
+`;
 
+const StopButton = styled.button`
+  background: ${props => props.theme.bottomBarButton};
+  font-size: 400%;
+  flex: 1;
   &:active {
     background: ${props => props.theme.bottomBarButtonActive};
   }
 `;
 
+const ConnectionBar = styled.div`
+  background-color: ${props =>
+    props.connecting
+      ? props.theme.notificationWarning
+      : props.connected
+        ? props.theme.notificationSuccess
+        : props.theme.notificationError};
```

Comment:
Why not create a function that maps isConnected to a color?

Reply:
sure